test/test_grocy.py

```diff
@@ -121,4 +122,74 @@ def test_get_stock_invalid_missing_data(self):
             }
         ]
         responses.add(responses.GET, "https://example.com/api/stock", json=resp, status=200)
+        
+    @responses.activate
+    def test_get_shopping_list_valid(self):
+        resp = [
+            {
+                "id": 1,
+                "product_id": 6,
+                "note": "string",
+                "amount": 2,
+                "row_created_timestamp": "2019-04-17 10:30:00",
+                "shopping_list_id": 1,
+                "done": 0
+            }
+        ]
+        responses.add(responses.GET, "https://example.com/api/objects/shopping_list", json=resp, status=200)
+
+        shopping_list = self.grocy.shopping_list()
+        
+        assert isinstance(shopping_list, list)
+        assert len(shopping_list) == 1
+        for item in shopping_list:
+            assert isinstance(item, ShoppingListProduct)
+            
+    @responses.activate
+    def test_get_shopping_list_invalid_no_data(self):
+        resp = [
```

Comment:
Currently this test case is the same as `test_get_shopping_list_invalid_missing_data`. Please change this to just include no response like in `test_get_stock_invalid_no_data`